API/Services/SchoolWorkerService.cs

```diff
@@ -0,0 +1,137 @@
+using System.Reflection;
+using System.Text.Json;
+using API.DTOs;
+using Microsoft.AspNetCore.Mvc;
+using CommonClasses;
+using CommonClasses.Enums;
+using CommonClasses.Models;
+
+namespace API.Services;
+
+public class SchoolWorkerService : WorkerBase
+{
+    public async Task<bool> CheckIfTaskHasCompleted(string taskId)
+    {
+        int tries = 5;
+        string? taskResult = string.Empty;
+        while (tries >= 0 || taskResult == "Done")
+        {
+             taskResult = await _db.StringGetAsync(taskId);
+             tries--;
+             await Task.Delay(200);
+        }
+
+        return true;
+    }
+    /*
+    public async Task InsertTaskIntoQueue(string taskId,TaskType taskName,TaskData taskData)
+    {
+        TaskRedis currTask = new TaskRedis()
+        {
+            TaskId = taskId,
+            TaskName = taskName,
+            TaskData = taskData
+        };
+        await _db.ListRightPushAsync("TasksQueue", JsonSerializer.Serialize(currTask));
+       
+    }*/
+     public async Task InsertTaskIntoQueue(string taskId,TaskType taskName,School taskData)
```

Comment:
regarding naming
instead of writing InsertTaskIntoQueue
rename to InsertTaskIntoQueueAsync

also, don't forget to add cancellation tokens when you are working with async functions